### PR TITLE
Require akka 2.6

### PR DESCRIPTION
**Pull request by raboof**
This branch is intended to be the starting point for Akka gRPC 1.1.0. We expect to rebase it onto master periodially while we work towards 1.0.0 there.

Refs #751

### CODE REVIEW

**Comment by patriknw** [comment]
why is this needed this early? for compatibility checks?

**Comment by raboof** [comment]
yes, the plan would be to use this branch to make sure there are no API's in 1.0.0 that would later make it hard to require Akka 2.6 without breaking backwards compatibility in 1.1.0. If this branch turns out to be hard to maintain we can drop it and reintroduce it later or find another way.

File: project/Dependencies.scala [hunk 1]
```diff
@@ -9,7 +9,7 @@ object Dependencies {
     val scala212 = "2.12.10"
     val scala213 = "2.13.1"
 
-    val akka = "2.5.27"
+    val akka = "2.6.2"
```
Comment (patriknw) [inline project/Dependencies.scala]: hold this until 2.6.3 since 2.6.2 is dead on arrival

**Comment by ignasi35** [comment]
Did a rebase of this one on top of `master`. I've pushed it to https://github.com/ignasi35/akka-grpc/tree/require-akka-2.6 (to not `push --force` here yet).

File: project/Dependencies.scala (hunk 1 above)
Comment (ignasi35) [reply project/Dependencies.scala]: The PR is now using `2.6.4`. Is there anything we explicitly need from `2.6.4`? We'd be better off requiring _only_ `2.6.3`.  Hmm, I'll try that out.

Comment (ignasi35) [reply project/Dependencies.scala]: Ok, `2.6.4` is required because of, at least, `parasitic`.

**Comment by raboof** [comment]
Rebased and cleaned up a bit, I think this is now good to go.

File: interop-tests/src/test/scala/akka/grpc/scaladsl/LoadBalancingIntegrationSpec.scala [hunk 2]
```diff
@@ -55,7 +56,7 @@ class LoadBalancingIntegrationSpec extends AnyWordSpec with Matchers with Before
     }
 
     "re-discover endpoints on failure" in {
-      val service1materializer = akka.stream.ActorMaterializer.create(system)
+      val service1materializer = Materializer(system)
```
Comment (ennru) [inline interop-tests/src/test/scala/akka/grpc/scaladsl/LoadBalancingIntegrationSpec.scala:59]: Is there a relevant difference to `SystemMaterializer(system).createAdditionalSystemMaterializer()` used in `NonBalancingIntegrationSpec`?

File: plugin-tester-scala/src/main/scala/example/myapp/helloworld/PowerGreeterServiceImpl.scala [hunk 3]
```diff
@@ -13,9 +13,8 @@ import example.myapp.helloworld.grpc._
 
 import scala.concurrent.Future
 
-class PowerGreeterServiceImpl(materializer: Materializer) extends GreeterServicePowerApi {
+class PowerGreeterServiceImpl(implicit materializer: Materializer) extends GreeterServicePowerApi {
```
Comment (ennru) [inline plugin-tester-scala/src/main/scala/example/myapp/helloworld/PowerGreeterServiceImpl.scala:16]: Shouldn't this take an `ActorSystem` now?

File: project/Dependencies.scala [hunk 4]
```diff
@@ -7,7 +7,7 @@ import buildinfo.BuildInfo
 object Dependencies {
   object Versions {
     val scala212 = "2.12.11"
-    val scala213 = "2.13.1"
+    val scala213 = "2.13.2"
```
Comment (ennru) [inline project/Dependencies.scala:10]: Take the latest?

**Review by ennru — APPROVED** [review]
LGTM.

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**